### PR TITLE
Feat/iam commands

### PR DESCRIPTION
**Pull request by JGiola**
## What this PR is for?

This PR will add commands for listing all the various IAM entities for a Mia-Platform Console company.

### CODE REVIEW

File: docs/30_commands.md [hunk 1]
```diff
@@ -102,6 +102,99 @@ Available flags for the command:
 - `--insecure-skip-tls-verify`, to disallow the check the validity of the certificate of the remote endpoint
 - `--context`, to specify a different context from the currently selected one
 
+### iam
+
+The `company iam` subcommands are used for managin the RBAC permissions associated with a company. Only
```
Comment (davidebianchi) [inline docs/30_commands.md:107]:
```suggestion
The `company iam` subcommands are used for managing the RBAC permissions associated with a company. Only
```

File: docs/30_commands.md [hunk 2]
```diff
@@ -102,6 +102,99 @@ Available flags for the command:
...
+**Company Owners** can modify, add or remove RBAC authorization to the company.
+
+#### list
+
+The `company iam list` subcommand allows you to view the list of all the different identity associated with the Company
+specified in the current  context. The output will show the **names**, **types** and **permissions** associated with
```
Comment (davidebianchi) [inline docs/30_commands.md:113]:
```suggestion
specified in the current context. The output will show the **names**, **types** and **permissions** associated with
```

File: docs/30_commands.md [hunk 3]
```diff
@@ -102,6 +102,99 @@ Available flags for the command:
...
+
+```sh
+miactl company iam list users [flags]
+```
+
+Available flags for the command:
+
+- `--groups`, filter IAM entities to show only groups. Mutally exclusive with `users` and `serviceAccounts`
+- `--serviceAccounts`, filter IAM entities to show only service accounts. Mutally exclusive with `users` and `groups`
+- `--users`, filter IAM entities to show only users. Mutally exclusive with `groups` and `serviceAccounts`
```
Comment (davidebianchi) [inline docs/30_commands.md:148]: Are those options correct? This command is already scoped to users

File: docs/30_commands.md [hunk 4]
```diff
@@ -102,6 +102,99 @@ Available flags for the command:
...
+
+```sh
+miactl company iam list groups [flags]
+```
+
+Available flags for the command:
+
+- `--groups`, filter IAM entities to show only groups. Mutally exclusive with `users` and `serviceAccounts`
+- `--serviceAccounts`, filter IAM entities to show only service accounts. Mutally exclusive with `users` and `groups`
+- `--users`, filter IAM entities to show only users. Mutally exclusive with `groups` and `serviceAccounts`
```
Comment (davidebianchi) [inline docs/30_commands.md:169]: Same as above

File: docs/30_commands.md [hunk 5]
```diff
@@ -102,6 +102,99 @@ Available flags for the command:
...
+- `--endpoint`, to set the Console endpoint (default is `https://console.cloud.mia-platform.eu`)
+- `--certificate-authority`, to provide the path to a custom CA certificate
+- `--insecure-skip-tls-verify`, to disallow the check the validity of the certificate of the remote endpoint
+- `--context`, to specify a different context from the currently selected one
+- `--company-id`, to set the ID of the desired Company
+
+##### serviceaccounts
+
+The `company iam list serviceaccounts` subcommand allows you to view the list of all service accounts that are availabe
```
Comment (davidebianchi) [inline docs/30_commands.md:178]:
```suggestion
The `company iam list serviceaccounts` subcommand allows you to view the list of all service accounts that are available
```

File: docs/30_commands.md [hunk 6]
```diff
@@ -102,6 +102,99 @@ Available flags for the command:
...
+
+```sh
+miactl company iam list serviceaccounts [flags]
+```
+
+Available flags for the command:
+
+- `--groups`, filter IAM entities to show only groups. Mutally exclusive with `users` and `serviceAccounts`
+- `--serviceAccounts`, filter IAM entities to show only service accounts. Mutally exclusive with `users` and `groups`
+- `--users`, filter IAM entities to show only users. Mutally exclusive with `groups` and `serviceAccounts`
```
Comment (davidebianchi) [inline docs/30_commands.md:191]: Same as above

File: internal/cmd/company/iam/parsers.go [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+		transformedRoles = append(transformedRoles, readableRole(role))
+	}
+
+	return transformedRoles
+}
+
+func readableRole(role string) string {
+	switch role {
+	case "company-owner":
+		return "company owner"
```
Comment (davidebianchi) [inline internal/cmd/company/iam/parsers.go:130]: This should not be `Company Owner`

File: internal/cmd/company/iam/parsers.go [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+	case "project-admin":
+		return "project administrator"
```
Comment (davidebianchi) [inline internal/cmd/company/iam/parsers.go:132]: `Project Administrator`?

**Review by davidebianchi — CHANGES_REQUESTED** [review]

File: internal/cmd/company/iam/parsers.go (hunk 7 above)
Comment (JGiola) [reply internal/cmd/company/iam/parsers.go:130]: The correct casing is done when the value will be shown to the user using the i18n `text` library functions.

File: internal/cmd/company/iam/parsers.go (hunk 8 above)
Comment (JGiola) [reply internal/cmd/company/iam/parsers.go:132]: The correct casing is done when the value will be shown to the user using the i18n `text` library functions.

**Review by davidebianchi — APPROVED** [review]

**Pull request merged**